MovieApp/MovieApp/MovieApp/ViewModels/MovieListPageViewModel.cs

```diff
@@ -0,0 +1,157 @@
+﻿using CSharpFunctionalExtensions;
+using MovieApp.Models;
+using MovieApp.Services;
+using Prism.Navigation;
+using Prism.Services;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace MovieApp.ViewModels
+{
+    public class MovieListPageViewModel : BaseViewModel
+    {
+        private readonly INavigationService _navigationService;
+        private readonly IMovieService _movieService;
+        private readonly IPageDialogService _dialogService;
+        private int _page = 1;
+
+        public bool IsLoading { get; set; }
+
+        private ObservableCollection<Movie> _movies;
+        public ObservableCollection<Movie> Movies
+        {
+            get { return _movies; }
+            set { SetProperty(ref _movies, value); }
+        }
+
+        private string _currentSearch;
+
+        private string _searchText;
+        public string SearchText
+        {
+            get { return _searchText; }
+            set { SetProperty(ref _searchText, value); }
+        }
+
+        public ICommand NavigateToDetailsCommand => new Command<Movie>(async (Movie movie) => await NavigateToDetailsCommandExecute(movie));
+        public ICommand LoadMoreMoviesCommand => new Command(async () => await ExecuteLoadMoreMoviesCommand());
+        public ICommand ReloadMoviesCommand => new Command(async () => await ExecuteReloadMoviesCommand());
+        public ICommand SearchMoviesCommand => new Command(async () => await ExecuteSearchMoviesCommand());
+        public ICommand ReloadCommand => new Command(async () => await ReloadCommandExecute());
+
+        public MovieListPageViewModel(IMovieService movieService, INavigationService navigationService, IPageDialogService dialogService)
+        {
+            _navigationService = navigationService;
+            _movieService = movieService;
+            _dialogService = dialogService;
+            Title = "Browse Movies";
+
+            LoadInitialData();
+        }
+
+        public async void LoadInitialData()
+        {
+            IsBusy = true;
```

Comment:
It is a good practice to catch any exception inside an async void method. Also if there is any exception within the method, the IsBusy flag will not be set to false.